File: year2/systems-design-implementation/Lab9/SwimServer/src/main/java/repo/intf/Repository.java
package repo.intf;

import utils.Entity;

import java.util.List;

public interface Repository<ID, E> {

    E findOne(ID id);

    List<E> findAll();

    E add(E entity) throws Exception;

    E delete(ID id);

    E update(E entity) throws Exception;

}

File: year2/systems-design-implementation/Lab4/Lab4Sharp/Repo/Interfaces/Repository.cs
using Lab2Sharp.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Repo.Interfaces
{
    public interface Repository<ID, E> where E : Entity<ID>
    {
        E FindOne(ID id);
        IEnumerable<E> FindAll();
        E Add(E entity);
        E Delete(ID id);
        E Update(E entity);
    }
}
